src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,77 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * A class representing a deadline task.
+ */
+public class Deadline extends Task {
+    private static final DateTimeFormatter DATE_TIME_FORMAT = DateTimeFormatter.ofPattern("MMM dd yyyy");
+    protected String by;
+    protected LocalDate byDate;
+
+    /**
+     * A constructor for the deadline task.
+     * 
+     * @param description The description of the task.
+     * @param by The note represented by a string.
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+        this.byDate = null;
```

Comment:
`this` is not required as its not shadowed

Reply:
I added `this` for clarity as there is another constructor that has `byDate` as a parameter, and in my opinion, better indicates this is a object variable.

Reply:
I see, but I think its just part of our coding guideline that if it's not shadowed, then we should not use null (or use null if and only if its shadowed) unfortunately

Reply:
I see, I might have missed it